src/lib/arch/aarch64/Instruction.cc

```diff
@@ -37,14 +37,17 @@ Instruction::Instruction(const Architecture& architecture,
 InstructionException Instruction::getException() const { return exception_; }
 
 const span<Register> Instruction::getSourceRegisters() const {
-  return {const_cast<Register*>(sourceRegisters.data()), sourceRegisterCount};
+  return {const_cast<Register*>(sourceRegisters.data()),
+          sourceRegisters.size()};
 }
 
 const span<RegisterValue> Instruction::getSourceOperands() const {
   return {const_cast<RegisterValue*>(operands.data()), operands.size()};
 }
 
 const span<Register> Instruction::getDestinationRegisters() const {
+  // destinationRegisterCount used as there may be +n in destinationRegisters
```

Comment:
Yes, `cmp` instructions often alias to `sub(s)` where the destination register is `<W|X>ZR` and they have an implicit destination of `NZCV`. So, when fetching the destination registers we only want NZCV and so `destinationRegisterCount` is used (in decode, `destinationRegisterCount` in not incremented when a zero destination register is come across, another variable is used to ensure `results` is the correct size for execute).

Reply:
Don't we handle this sort of thing in the revertAliasing logic? Trying to figure out how we had this issue all along but no discernable functional effect in all our benchmarks.

Reply:
Perhaps this is some pre-development readying for a capstone update?

Reply:
As per I got my alias terminology the wrong way round..... cmp is an alias of subs, and so `<W|X>ZR` is added as a destination in revertAlias. 

Before this PR, when the SME additions were added (and currently viewable in `dev`) you'll see that the `results` and `operands` vectors always had `+1` added to their size when re-sizing. This was due to (at the time) unknown segfaults if this wasn't added. Said `+1` allowed for simulation to not fail, and returning `destinationRegisterCount` ensured only the correct number of registers is returned. Within this PR it was discovered that this segfault was due to how `XZR/WZR` destination registers were handled (i.e. they were ignored by decode which wasn't an issue with fixed size arrays, but was with vectors). 

Hence, with the vectors now being the correct size iff they have a WZR/XZR destination register (see instruction decode for how this is done), I thought this comment would be useful to indicate why we use `destinationRegisterCount` rather than just return the size of the vector.

Reply:
Okay, that makes a lot more sense. A slight rewording for clarity. Something like " The destinationRegisterCount is used here as the span count value because there may be n number of zero registers in the latter indexes of the destinationRegisters vector. These cannot be written to and hence shouldn't be included in the returned span"

Reply:
Have updated the comment